pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1541,6 +1788,86 @@ func (dr *DashboardServiceImpl) searchDashboardsThroughK8sRaw(ctx context.Contex
 	return ParseResults(res, 0)
 }
 
+type dashboardProvisioningWithUID struct {
+	dashboards.DashboardProvisioning
+	DashboardUID string
+}
+
+func (dr *DashboardServiceImpl) searchProvisionedDashboardsThroughK8s(ctx context.Context, query dashboards.FindPersistedDashboardsQuery) ([]*dashboardProvisioningWithUID, error) {
+	ctx = identity.WithRequester(ctx, getDashboardBackgroundRequester(query.OrgId))
+
+	if query.ProvisionedRepo != "" {
+		query.ProvisionedRepo = provisionedFileNameWithPrefix(query.ProvisionedRepo)
+	}
+
+	if len(query.ProvisionedReposNotIn) > 0 {
+		repos := make([]string, len(query.ProvisionedReposNotIn))
+		for i, v := range query.ProvisionedReposNotIn {
+			repos[i] = provisionedFileNameWithPrefix(v)
+		}
+		query.ProvisionedReposNotIn = repos
+	}
+
+	searchResults, err := dr.searchDashboardsThroughK8sRaw(ctx, &query)
+	if err != nil {
+		return nil, err
+	}
+
+	newCtx, cancel, err := dr.getK8sContext(ctx)
+	if err != nil {
+		return nil, err
+	} else if cancel != nil {
+		defer cancel()
+	}
+
+	client, ok := dr.k8sclient.getClient(newCtx, query.OrgId)
+	if !ok {
+		return nil, nil
+	}
+
+	dashs := make([]*dashboardProvisioningWithUID, 0)
+	for _, hit := range searchResults.Hits {
```

Comment:
NIT – we could also use `errgroup.WithContext` for concurrent fetching here.

Reply:
Good idea - updated!